src/main/commandServer/__tests__/settingsValidator.spec.ts

```diff
@@ -0,0 +1,150 @@
+const subject = new SettingsValidator(cfg);
+
+subject.k8sVersions = [finalK8sVersion, '1.0.0'];
+describe(SettingsValidator, () => {
+  describe('validateSettings', () => {
+    it('should do nothing when given existing settings', () => {
+      const [needToUpdate, errors] = subject.validateSettings(cfg);
+
+      expect(needToUpdate).toBeFalsy();
+      expect(errors.length).toEqual(0);
+    });
+
+    it('should want to apply changes when valid new settings are proposed', () => {
+      const newEnabled = !cfg.kubernetes.enabled;
+      const newVersion = subject.k8sVersions[1];
+      const newEngine = cfg.kubernetes.containerEngine === 'moby' ? 'containerd' : 'moby';
+      const newConfig = _.merge({}, cfg, {
+        kubernetes:
+          {
+            enabled:         newEnabled,
+            version:         newVersion,
+            containerEngine: newEngine
+          }
+      });
+      const [needToUpdate, errors] = subject.validateSettings(newConfig);
+
+      expect(needToUpdate).toBeTruthy();
+      expect(errors.length).toEqual(0);
+    });
+
+    it('should report errors for unchangeable fields', () => {
+      const desiredEnabled = !cfg.kubernetes.enabled;
+      const desiredEngine = cfg.kubernetes.containerEngine === 'moby' ? 'containerd' : 'moby';
+      const requestedSettings = {
+        kubernetes:
+          {
+            enabled:                    desiredEnabled,
+            containerEngine:            desiredEngine,
+            checkForExistingKimBuilder: !cfg.kubernetes.checkForExistingKimBuilder,
+          }
+      };
+      const [needToUpdate, errors] = subject.validateSettings(requestedSettings);
+
+      expect(needToUpdate).toBeFalsy();
+      expect(errors.length).toEqual(1);
+      expect(errors).toContain("Changing field kubernetes.checkForExistingKimBuilder via the API isn't supported.");
+    });
+
+    it('should report errors for unchangeable fields when all settings are present', () => {
+      const desiredEnabled = !cfg.kubernetes.enabled;
+      const desiredEngine = cfg.kubernetes.containerEngine === 'moby' ? 'containerd' : 'moby';
+      const requestedSettings = _.merge({}, cfg, {
+        kubernetes:
+          {
+            enabled:                    desiredEnabled,
+            containerEngine:            desiredEngine,
+            checkForExistingKimBuilder: !cfg.kubernetes.checkForExistingKimBuilder,
+          }
+      });
+      const [needToUpdate, errors] = subject.validateSettings(requestedSettings);
+
+      expect(needToUpdate).toBeFalsy();
+      expect(errors.length).toEqual(1);
+      expect(errors[0]).toContain("Changing field kubernetes.checkForExistingKimBuilder via the API isn't supported.");
+    });
+
+    it('should complain about all unchangeable fields', () => {
+      const valuesToChange: [RecursivePartial<settings.Settings>, string][] = [
+        [{ version: cfg.version + 1 }, 'version'],
```

Comment:
Would be nice if we could typecheck the fully qualified name… but it's not worth it, it'll be a test failure anyway (as `Setting name X isn't recognized` instead of `Changing field X via the API isn't supported.`).

Reply:
Nothing to do here.